File: app/src/main/java/com/gmail/wondergab12/restapi/dependencies/ApplicationModule.java
package com.gmail.wondergab12.restapi.dependencies;

import android.app.Application;

import androidx.lifecycle.ViewModelProvider;

import com.gmail.wondergab12.restapi.vm.MainViewModel;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@SuppressWarnings("WeakerAccess")
@Module
public class ApplicationModule {

    @Singleton
    @Provides
    public MainViewModel provideMainViewModel(Application application) {
        return ViewModelProvider.AndroidViewModelFactory
                .getInstance(application)
                .create(MainViewModel.class);
    }
}

File: app/src/main/java/com/gmail/wondergab12/restapi/dependencies/ApplicationModule.kt
package com.gmail.wondergab12.restapi.dependencies

import android.app.Application
import androidx.lifecycle.ViewModelProvider
import com.gmail.wondergab12.restapi.vm.MainViewModel
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule {

    @ApplicationScope
    @Provides
    fun provideMainViewModel(application: Application): MainViewModel {
        return ViewModelProvider.AndroidViewModelFactory
                .getInstance(application)
                .create(MainViewModel::class.java)
    }
}
